src/XComponentAPI.js

```diff
@@ -0,0 +1,70 @@
+
+define(["Connection", "Parser", "GetterXcApi"], function (Connection, Parser, GetterXcApi) {
+    "use strict";
+
+    var Factory = function () {
+        this.createXComponentApi = function (url) {
+            var xcApiString = GetterXcApi.Getter();
+            var parser = new Parser(xcApiString);
+            var factoryConnection = new Connection.Factory();
+            var connection = factoryConnection.createConnection(url);
+            var api = new XComponentAPI(connection, parser);
+            return api;
+        }
+    }
+
+
+    var XComponentAPI = function(connection, parser) {
+
+        var _connection = connection;
+
+        this.getConnection = function () {
+            return _connection;
+        }
+
+        var _parser = parser;
+
+        this.getParser = function () {
+            return _parser;
+        }
+
+    }
+    
+    
+    XComponentAPI.prototype.getEventToSend = function (componentName, stateMachineName, jsonMessage) {
```

Comment:
better name createEventToSend

It should be a private method.
Pershaps a better idea is to create an EventConverter class...